src/time_integration/time_integration.jl

```diff
@@ -82,6 +84,15 @@ function DiffEqBase.get_tstops_max(integrator::AbstractTimeIntegrator)
     return maximum(get_tstops_array(integrator))
 end
 
+function unstable_check(dt, u_ode, integrator::AbstractTimeIntegrator, t)
+    if mpi_isparallel()
+        u_isfinite = MPI.Allreduce!(Ref(all(isfinite, u_ode)), Base.min, mpi_comm())[]
+    else
+        u_isfinite = all(isfinite, u_ode)
+    end
+    return !isfinite(dt) || !u_isfinite
+end
+
```

Comment:
Since this function is potentially called in each stage (or at least each step), I'd refrain from making it the default, since an allreduce is not for free for parallel simulations.

Can we make it optional instead?

Reply:
(or is this already the default elsewhere?)

Reply:
Also, why does the current `ode_unstable_check` not suffice?

Reply:
No, I agree that it is not good to have a simulation fail "silently"! However, I'm confused: Why does the current `ode_unstable_check` fail to detect the NaNs? IIRC, it just checks if `dt` is NaN, and since `dt` depends on the current solution, it should also be `NaN`?

Is it that the time step size is not computed in each time step? Or are we dealing with some sort of "non-propagating" NaNs?

Reply:
Firstly, I just realized that `ode_unstable_check` isn't even called (even if used as a kwarg in `solve`), since there is no kwarg `unstable_check` in the custom integrators.

But to your questions. `dt` is calculated every time step and `lambda1` or `lambda2` are `NaN` at some points, but we use the Trixi-intern `max` function with
```
julia> max(1.1, NaN)
NaN

julia> Trixi.max(1.1, NaN)
1.1
```

Reply:
Ah, thanks for the clarification :=) @bennibolm Can you please use the `Base.max` version there as suggested, together with an appropriate comment on why we use it here? That would IMHO the most elegant solution.

Reply:
Yes, I believe this would make sense (unless @ranocha objects)

Reply:
See https://github.com/trixi-framework/Trixi.jl/pull/2566